homeassistant/components/zwave_js/triggers/event.py

```diff
@@ -145,7 +142,12 @@ async def async_attach_trigger(
     platform_type: str = PLATFORM_TYPE,
 ) -> CALLBACK_TYPE:
     """Listen for state changes based on configuration."""
-    nodes: set[Node] = config.get(ATTR_NODES, {})
+    dev_reg = dr.async_get(hass)
+    nodes = async_get_nodes_from_targets(hass, config, dev_reg=dev_reg)
+    if config[ATTR_EVENT_SOURCE] == "node" and not nodes:
+        raise vol.Invalid(
```

Comment:
Is it ok to raise `vol.Invalid` here? We're not inside config validation anymore, I think.

Reply:
Oops I can raise a better exception. Are we ok raising an exception here? The automation doesn't get loaded and the error appears in the logs

Reply:
I'm not sure what our protocol is here.

Normally if the user input is wrong to a service we raise ValueError in the service handler. This isn't exactly that case, but maybe close enough? Look around if there are some other examples.

Reply:
Should we change to raise a ValueError here too?

Reply:
Yup thanks for catching that